packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/Abs_L2DepositedToken.sol

```diff
@@ -139,66 +139,95 @@ abstract contract Abs_L2DepositedToken is iOVM_L2DepositedToken, OVM_CrossDomain
 
     /**
      * @dev initiate a withdraw of some tokens to the caller's account on L1
-     * @param _amount Amount of the token to withdraw
+     * @param _amount Amount of the token to withdraw.
+     * @param _data Data to forward to L1.
+     * @param _l1Gas Gas limit for the provided message.
      */
     function withdraw(
-        uint _amount
+        uint _amount,
+        bytes calldata _data,
+        uint32 _l1Gas
     )
         external
         override
         virtual
         onlyInitialized()
     {
-        _initiateWithdrawal(msg.sender, _amount);
+        _initiateWithdrawal(
+            msg.sender,
+            msg.sender,
+            _amount,
+            _data,
+            _l1Gas
+        );
     }
 
     /**
-     * @dev initiate a withdraw of some token to a recipient's account on L1
-     * @param _to L1 adress to credit the withdrawal to
-     * @param _amount Amount of the token to withdraw
+     * @dev initiate a withdraw of some token to a recipient's account on L1.
+     * @param _to L1 adress to credit the withdrawal to.
+     * @param _amount Amount of the token to withdraw.
+     * @param _data Data to forward to L1.
+     * @param _l1Gas Gas limit for the provided message.
      */
     function withdrawTo(
         address _to,
-        uint _amount
+        uint _amount,
+        bytes calldata _data,
+        uint32 _l1Gas
     )
         external
         override
         virtual
         onlyInitialized()
     {
-        _initiateWithdrawal(_to, _amount);
+        _initiateWithdrawal(
+            msg.sender,
+            _to,
+            _amount,
+            _data,
+            _l1Gas
+        );
     }
 
     /**
      * @dev Performs the logic for deposits by storing the token and informing the L2 token Gateway of the deposit.
      *
-     * @param _to Account to give the withdrawal to on L1
-     * @param _amount Amount of the token to withdraw
+     * @param _from Account to pull the deposit from on L2.
+     * @param _to Account to give the withdrawal to on L1.
+     * @param _amount Amount of the token to withdraw.
+     * @param _data Data to forward to L1.
+     * @param _l1Gas Gas limit for the provided message on L1.
      */
     function _initiateWithdrawal(
+        address _from,
         address _to,
-        uint _amount
+        uint _amount,
+        bytes calldata _data,
+        uint32 _l1Gas
     )
         internal
     {
         // Call our withdrawal accounting handler implemented by child contracts (usually a _burn)
         _handleInitiateWithdrawal(_to, _amount);
 
         // Construct calldata for l1TokenGateway.finalizeWithdrawal(_to, _amount)
-        bytes memory data = abi.encodeWithSelector(
+        bytes memory message = abi.encodeWithSelector(
```

Comment:
`message` is consistent with the variable naming in the Cross Domain Messengers:

https://github.com/ethereum-optimism/optimism/blob/b0b45b6816771ec2bc2f8f4d7574308aa224c9db/packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/Abs_BaseCrossDomainMessenger.sol#L69-L73

and avoids conflict with the new `_data` argument.

Reply:
I like this convention, will try to stick with it more myself :)